Bottom: ad711dcbfa681503c16b448c5dc94d9ee4d24ff2
Top:    855f32668e9d36a4cc6211b954c60df0b0bd7f51
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-15 15:04:51 +0800

Refresh of virtio-net

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 7509cd4..02329bf 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -813,7 +813,7 @@ bool virtnet_frag_refill(struct receive_queue *rq, unsigned int sz,
 	}
 
 	pfrag->offset = 0;
-	if (gfp == GFP_ATOMIC && rq->page_cache.index > 0) {
+	if ((gfp & GFP_ATOMIC) && rq->page_cache.index > 0) {
 		pfrag->page = rq->page_cache.pages[--rq->page_cache.index];
 		pfrag->size = PAGE_SIZE << compound_order(pfrag->page);
 		return true;
